pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoParserTools.java

```diff
@@ -216,8 +217,11 @@ public static double readDouble(final ReadableSequentialData input) throws IOExc
      */
     public static String readString(final ReadableSequentialData input) throws IOException {
         final int length = input.readVarInt(false);
+        if (input.remaining() < length) {
```

Comment:
I think this check is not needed as it as `readBytes` will only read remaining bytes and return a read number smaller than length which will be caught by second `if`

Reply:
See the explanation above. I changed it to be a pre-read check and fast fail. This way the code runs faster too and is consistent with how exceptions are thrown in BufferedData and the rest of ReadableSequentialData.